### ISSUE TITLE
Support for transactions

### ISSUE THREAD

**Issue by gvolpe** [zmactep/hasbolt#14]
Hi @zmactep , thanks for making this library!

I see that project looks a bit abandoned and I was wondering if you have thought at any point about introducing support for transactions? 

I could give it a shot and see if I can implement it if you think it could be merged :)

**Comment by zmactep** [comment]
Hi @gvolpe!

I thing transaction implementation is a great idea (really, do not understand why I haven't done it before). It would be great if you'll try to implement it. :)

**Comment by gvolpe** [comment]
Awesome, I'll give it a shot, thanks :)

**Comment by zmactep** [comment]
The solution at #16 supports this kind of syntax:
```haskell
query1 :: MonadIO m => BoltActionT m [Record]
query1 = query "CREATE (g:Guitar { brand: 'Gibson' } ) RETURN g"

query2 :: MonadIO m => BoltActionT m ()
query2 = query_ "CREATE (g:Guitar { brand: 'Fender' } ) RETURN g"
 
query3 :: MonadIO m => BoltActionT m Int
query3 = length <$> query "CREATE (g:Guitar { brand: 'Ibanez' } ) RETURN g"

query4 :: MonadIO m => BoltActionT m Record
query4 = head <$> query "This is going to make it fail, BOOM!"

query5 :: MonadIO m => BoltActionT m [Record]
query5 = query "CREATE (g:Guitar { brand: 'Schecter' } ) RETURN g"

-- We can combine different types of queries or even perform some actions inside
transaction :: MonadIO m => BoltActionT m [Record]
transaction = transact $ do
    records <- query1
    query2
    number <- query3
    forM_ [0..number] $ \_ ->
      query4
    query5
```

The transaction would be rolled back in case of wrong cypher request or any other failure inside the `transact` block (e.g. wrong data unpacking).

**Comment by gvolpe** [comment]
Looks great @zmactep ! It'd be nice if you could include some unit tests. Maybe take the ones from here or if you prefer merge #16 and then I can PR the tests.

**Comment by zmactep** [comment]
Tests could be found in new commit to #16. They are disabled by default as they require neo4j at localhost with `neo4j/test` credentials.

**Comment by gvolpe** [comment]
Fixed by #16
